class.jetpack-gutenberg.php

```diff
@@ -16,63 +54,56 @@ public static function is_gutenberg_available() {
 	}
 
 	/**
-	 * Load Gutenberg assets
+	 * Check whether conditions indicate Gutenberg blocks should be loaded
+	 *
+	 * Loading blocks is enabled by default and may be disabled via filter:
+	 *   add_filter( 'jetpack_gutenberg', '__return_false' );
 	 *
 	 * @since 6.7.0
 	 *
-	 * @return void
+	 * @return bool
 	 */
-	public static function enqueue_block_assets() {
-		if ( ! self::should_load_blocks() ) {
-			return;
+	public static function should_load_blocks() {
+		if ( ! Jetpack::is_active() ) {
+			return false;
 		}
 
-		$rtl = is_rtl() ? '.rtl' : '';
-
 		/**
-		 * Filter to enable serving blocks via CDN
-		 *
-		 * CDN cache is busted once a day or when Jetpack version changes. To customize it:
-		 *   add_filter( 'jetpack_gutenberg_cdn_cache_buster', function( $version ) { return time(); }, 10, 1 );
+		 * Filter to disable Gutenberg blocks
 		 *
 		 * @since 6.5.0
 		 *
-		 * @param bool false Whether to load Gutenberg blocks from CDN
+		 * @param bool true Whether to load Gutenberg blocks
 		 */
-		if ( apply_filters( 'jetpack_gutenberg_cdn', false ) ) {
-			$cdn_base    = 'https://s0.wp.com/wp-content/mu-plugins/jetpack/_inc/blocks';
-			$view_script = "$cdn_base/view.js";
-			$view_style  = "$cdn_base/view$rtl.css";
-
-			/**
-			 * Filter to modify cache busting for Gutenberg block assets loaded from CDN
-			 *
-			 * @since 6.5.0
-			 *
-			 * @param string
-			 */
-			$version = apply_filters( 'jetpack_gutenberg_cdn_cache_buster', sprintf( '%s-%s', gmdate( 'd-m-Y' ), JETPACK__VERSION ) );
-		} else {
-			$view_script = plugins_url( '_inc/blocks/view.js', JETPACK__PLUGIN_FILE );
-			$view_style  = plugins_url( "_inc/blocks/view$rtl.css", JETPACK__PLUGIN_FILE );
-			$version     = Jetpack::is_development_version() && file_exists( JETPACK__PLUGIN_DIR . '_inc/blocks/view.js' )
-				? filemtime( JETPACK__PLUGIN_DIR . '_inc/blocks/view.js' )
-				: JETPACK__VERSION;
+		return (bool) apply_filters( 'jetpack_gutenberg', true );
+	}
+
+	public static function load_assets_as_required( $type ) {
```

Comment:
Do we want to make sure that the `$type` is slug-like before using it? Perhaps passing it to `sanitize_title_with_dashes()` or `sanitize_title()` should be enough.

Reply:
I think we are OK with not so much checking since this is only supposed to be used from withing code. And I expect code review to be able to catch such things as well.

Reply:
I agree, but also any third party can call `jetpack_register_block()` or `Jetpack_Gutenberg::add_block()`, right? Never can be too safe 😉 

Anyway, I'll leave this to your judgement.